src/main/java/Eddie/Parser.java

```diff
@@ -0,0 +1,119 @@
+package Eddie;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.util.Scanner;
+
+/**
+ * Represents a parser which makes sense of the user input.
+ */
+public class Parser {
+    /**
+     * Main control flow for recognizing commands and what to do with commands.
+     * @throws DukeException Thrown when commands are not inputted correctly.
+     */
+    private static String execute(String s) throws DukeException {
+        Scanner sc = new Scanner(s);
+        while (true) {
+
+            String command = sc.next();
+
+
+            if (command.equals("bye")) {
+                return Duke.exit();
+            } else if (command.equals("list")) {
+                return Duke.list();
+            } else if (command.equals("mark")) {
+                int taskNum = sc.nextInt();
+                Task task = TaskList.get(taskNum - 1);
+                task.taskIsDone();
+                Storage.write();
+                return Ui.mark(taskNum);
+            } else if (command.equals("unmark")) {
+                int taskNum = sc.nextInt();
+                Task task = TaskList.get(taskNum - 1);
+                task.taskNotDone();
+
+                Storage.write();
+                return Ui.unmark(taskNum);
+            } else if (command.equals("delete")) {
+                int index = sc.nextInt();
+                return Duke.delete(index - 1);
+            } else if (command.equals("todo")) {
+
+                String restOfString = sc.nextLine();
+                if (restOfString.length() != 0) {
+                    String taskName = restOfString;
+                    Task taskToAdd = new Todo(taskName);
+                    return Duke.add(taskToAdd);
+
+                } else {
+                    throw new EmptyDescriptionException();
+                }
```

Comment:
Small nitpick but is there a reason for leaving empty lines in the middle here?

Reply:
Oops seems like I deleted something but didnt remove the line